File: SDS/numberTheory/num_G_1837.py
P, K = map(int, input().split())

MAX = 1000000

primes = [ True ] * (MAX+1)

find = False

for i in range(2, int(MAX**0.5)+1):
#for i in range(2, MAX+1):    
    if primes[i]:
        for j in range(i+i, MAX+1, i):
            primes[j] = False

# pq, p or q < K

#print(primes)

for k in range(2, K):
    if primes[k]:
        if P % k == 0:
            find = True
            print("BAD", k)
            break
            
if not find:
    print("GOOD")

File: SDS/numberTheory/num_G_1837.cpp
#include<iostream>
#include<vector>
#include<math.h>

using namespace std;

#define MAX 100001

bool primes[MAX];

void che()
{
    for(int i=2; i < int(sqrt(MAX))+1 ; i++)   
    {
        if(!primes[i])
        {
            for(int j=i*i; j <= MAX ; j += i)
                primes[j] = true;
        }
    }
}

int main()
{
    int P, K;
    int find = 0;
    scanf("%d %d", &P, &K);
    
    for(int i=2; i<= K ; i++)
    {
        if(!primes[i])
        {
            if(P % i == 0)
            {
                find = 1;
                printf("BAD %d\n", i);
                break;
            }
        }
    }
        
    if(!find)
        printf("GOOD\n");
}